pandas/core/indexes/datetimes.py

```diff
@@ -901,6 +903,11 @@ def _sub_datelike_dti(self, other):
             new_values[mask] = libts.iNaT
         return new_values.view('i8')
 
+    @Substitution(klass='DatetimeIndex')
+    @Appender(DatetimeIndexOpsMixin.shift.__doc__)
```

Comment:
this should be a shared doc-string instead (and same for the other things you added). this is the existing pattern.